In [6]:
library(powerSurvEpi)
options(warn=-1)

### Univariate Analysis

From the paper, 

| Stages        | No Surgery (%)| Surgery (%)|
| ------------- |---------------| -----------|
| I             | 66.2          | 33.8       |
| II            | 79.7          | 20.3       |
| III           | 96.7          | 3.3        |

| Stages        | OS - Hazard ratio of surgery | LCSS - Hazard ratio of surgery |
| ------------- |------------------------------| -------------------------------|
| I             | 0.369                        | 0.335                          |
| II            | 0.549                        | 0.506                          |
| III           | 0.477                        | 0.456                          |

For stage I patients, the cumulative OS rate was 2.9% for no-surgery versus 21.0% for surgery, and the LCSS rate was 8.5% for no-surgery versus 43.2% for the surgery group. 

For stage II patients, the cumulative OS rate was 2.0% for no-surgery compared to 5.9% for surgery, and the LCSS rate was 5.5% for no-surgery compared to 20.4% for the surgery group. 

For stage III patients, the cumulative OS rate was 0.9% for no-surgery versus 5.1% for surgery, and the LCSS rate was 3.0% for no-surgery versus 8.0% for the surgery group. 

Arguments of <b>ssizeCT.default</b>:
- power - power to detect the magnitude of the hazard ratio as small as that specified by RR. 
- k - ratio of participants in group E (experimental group) compared to group C (control group). 
- pE - probability of failure in group E (experimental group) over the maximum time period of the study (t years).
- pC - probability of failure in group C (control group) over the maximum time period of the study (t years). 
- RR - postulated hazard ratio. 
- alpha - type I error rate.

### Sample size calculation for the Comparison of Survival Curves Between Two Groups under the Cox Proportional-Hazards Model for clinical trials.

In [15]:
# For stage I OS univariate analysis, experimental group is the group who had surgery
ssizeCT.default(power = 0.95, k = 33.8/66.2, pE = (1 - 0.21), pC = (1 - 0.029), RR = 0.369, alpha = 0.05)

nE nC 
34 66

In [17]:
# For stage I LCSS univariate analysis, experimental group is the group who had surgery
ssizeCT.default(power = 0.95, k = 33.8/66.2, pE = (1 - 0.432), pC = (1 - 0.085), RR = 0.335, alpha = 0.05)

nE nC 
34 66

In [25]:
# For stage II OS univariate analysis, experimental group is the group who had surgery
ssizeCT.default(power = 0.95, k = 20.3/79.7, pE = (1 - 0.059), pC = (1 - 0.02), RR = 0.549, alpha = 0.05)

nE  nC 
 69 268

In [23]:
# For stage II LCSS univariate analysis, experimental group is the group who had surgery
ssizeCT.default(power = 0.95, k = 20.3/79.7, pE = (1 - 0.204), pC = (1 - 0.055), RR = 0.506, alpha = 0.05)

nE  nC 
 60 233

In [20]:
# For stage III OS univariate analysis, experimental group is the group who had surgery
ssizeCT.default(power = 0.95, k = 3.3/96.7, pE = (1 - 0.051), pC = (1 - 0.009), RR = 0.477, alpha = 0.05)

nE   nC 
  48 1405

In [21]:
# For stage III LCSS univariate analysis, experimental group is the group who had surgery
ssizeCT.default(power = 0.95, k = 3.3/96.7, pE = (1 - 0.08), pC = (1 - 0.03), RR = 0.456, alpha = 0.05)

nE   nC 
  46 1326

### Summary of results

<head>
<style>
table {
    font-family: arial, sans-serif;
    border-collapse: collapse;
    width: 100%;
}

td, th {
    border: 2px solid #dddddd;
    text-align: left;
    padding: 8px;
}

tr:nth-child(even) {
    background-color: #dddddd;
}
</style>
</head>

<body>
<table align="left">
  <tr>
  	<td><b>Stages</b></td>
    <td colspan="2"><b>OS</b></td>
    <td colspan="2"><b>LCSS</b></td>
  </tr>
  <tr>
  	<td></td>
    <td>Surgery</td>
    <td>No Surgery</td>
    <td>Surgery</td>
    <td>No Surgery</td>
  </tr>
  <tr>
  	<td><b>I</b></td>
    <td>34(344)</td>
    <td>66(674)</td>
    <td>34(344)</td>
    <td>66(674)</td>
  </tr>
  <tr>
  	<td><b>II</b></td>
    <td bgcolor="#F37676">69(60)</td>
    <td bgcolor="#F37676">268(235)</td>
    <td>60(60)</td>
    <td>233(235)</td>
  </tr>
  <tr>
  	<td><b>III</b></td>
    <td>48(116)</td>
    <td>1405(3351)</td>
    <td>46(116)</td>
    <td>1326(3351)</td>
  </tr>
</table>
</body>